supervisor/api/resolution.py

```diff
@@ -62,3 +78,17 @@ async def dismiss_issue(self, request: web.Request) -> None:
     def healthcheck(self, request: web.Request) -> Awaitable[None]:
         """Run backend healthcheck."""
         return asyncio.shield(self.sys_resolution.healthcheck())
+
+    @api_process
+    @require_home_assistant
+    async def check_options(self, request: web.Request) -> None:
+        """Set check options."""
+        body = await api_validate(SCHEMA_CHECK_OPTIONS, request)
```

Comment:
Should we make this a decorator? 😄 

https://github.com/home-assistant/core/blob/dev/homeassistant/components/http/data_validator.py#L16

Reply:
not in this PR

Reply:
btw any reason you're keeping the schema defined on the top of the file? Doesn't that make it harder to follow what's going on because now the schema and the logic are on the other side of the page?

Reply:
That's the default :shrug:

Reply:
Challenge everything.

Reply:
Yeah, there like the API handling and registration process need a rework